File: Leetcode/24-Swap-Nodes-in-Pairs/SwapNodesInPairs.js
/**
 * Definition for singly-linked list.
 * function ListNode(val) {
 *     this.val = val;
 *     this.next = null;
 * }
 */
/**
 * @param {ListNode} head
 * @return {ListNode}
 */
var swapPairs = function(head) {
    if (!head || !head.next) {
        return head
    }
    var dummy = new ListNode(0)
    dummy.next = head
    var A = head
    head = A.next
    var B
    while (A) {
        B = A.next
        if (!B) {
            break
        }
        A.next = B.next
        B.next = A
        dummy.next = B
        dummy = A
        A = A.next
    }
    return head
};

File: Leetcode/24-Swap-Nodes-in-Pairs/SwapNodesInPairs.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode(int x) { val = x; }
 * }
 */
public class SwapNodesInPairs {
    public ListNode swapPairs(ListNode head) {
        if (head == null || head.next == null) {
            return head;
        }
        ListNode dummy = new ListNode(0);
        ListNode A = head;
        dummy.next = A;
        head = A.next;
        ListNode B;
        while (A!=null) {
            B = A.next;
            if (B == null)
                break;
            
            A.next = B.next;
            B.next = A;
            dummy.next = B;
            dummy = A;
            A = A.next;
        }
        return head;
    }
}